account_financial_report_webkit/report/webkit_parser_header_fix.py

```diff
@@ -149,8 +149,29 @@ def generate_pdf(self, comm_path, report_xml, header, footer, html_list,
         command.append(out_filename)
         stderr_fd, stderr_path = tempfile.mkstemp(text=True)
         file_to_del.append(stderr_path)
+
+        # The encoding can break subprocess in some Python versions,
+        # ensure that all args are in ASCII or Unicode.
+        encoding_parsers = [
+            lambda s: s,
+            lambda s: s.decode('utf-8-sig') if isinstance(s, str) else s,
+            lambda s: s.encode('utf-8-sig') if isinstance(s, unicode) else s,
+            lambda s: s.decode('utf-8') if isinstance(s, str) else s,
+            lambda s: s.encode('utf-8') if isinstance(s, unicode) else s,
+        ]
+
         try:
-            status = subprocess.call(command, stderr=stderr_fd)
+            for encoding_parser in encoding_parsers:
+                try:
+                    comm = [encoding_parser(c) for c in command]
+                    _logger.debug('Running command: %r', comm)
+                    status = subprocess.call(comm, stderr=stderr_fd)
+                except Exception as e:
```

Comment:
Well the problem is that all the elements in the arguments list are always `strings` –as expected– and that this issue appears after some time of use, which makes this bug so difficult to reproduce.

See the next commands list example –it sometimes works, sometimes doesn't–:
> ['/usr/local/bin/wkhtmltopdf', '--quiet', '--encoding', 'utf-8', '--orientation', 'Landscape', '--page-size', 'A4', '--header-font-name', 'Helvetica', '--footer-font-name', 'Helvetica', '--header-font-size', '10', '--footer-font-size', '6', '--header-left', u'LIBRO MAYOR - My Company, S.L. - EUR', '--header-spacing', '2', '--footer-left', '13/05/2015 15:10:35', '--footer-right', u'P\xe1gina [page] de [topage]', '--footer-line', '/tmp/tmpNJ0AlL0.body.html', '/tmp/webkit.tmp.2mzEzV.pdf']

The only clue is that setting a different user's default language seems to fix it (e.g. from **es** to **en_GB** or **es_CL**) so it may be a translations-related issue with **subprocess**, as the translated terms seems to be in *Unicode* –see the `u''`– instead of *ASCII*. But –as mentioned– it sometimes works "out of the box"...

Clearing, building and/or overwriting the translation terms doesn't always helps fixing this bug, only forcing the encoding as shown.

PS: If it helps, we're using `Debian GNU/Linux 7.8 (wheezy)` with `es_ES.UTF-8` as the **openerp** user locale, and `Python 2.7.3` with the next code snippet ir the `sitecustomize.py`:
```
import sys
sys.getdefaultencoding('utf-8')
```